src/Symfony/Component/Validator/Tests/Constraints/BicValidatorTest.php

```diff
@@ -36,6 +57,113 @@ public function testEmptyStringIsValid()
+    {
+        $constraint = new Bic(array('ibanPropertyPath' => '[root][value]'));
+
+        $this->setObject(array('root' => array('value' => 'FR14 2004 1010 0505 0001 3M02 606')));
+
+        $this->validator->validate('SOGEFRPP', $constraint);
+
+        $this->assertNoViolation();
+    }
+
+    public function testInvalidComparisonToPropertyPath()
+    {
+        $constraint = new Bic(array('ibanPropertyPath' => 'value'));
+        $constraint->ibanMessage = 'Constraint Message';
+
+        $object = new BicComparisonTest_Class('FR14 2004 1010 0505 0001 3M02 606');
+
+        $this->setObject($object);
+
+        $this->validator->validate('UNCRIT2B912', $constraint);
+
+        $this->buildViolation('Constraint Message')
+            ->setParameter('{{ value }}', '"UNCRIT2B912"')
+            ->setParameter('{{ iban }}', 'FR14 2004 1010 0505 0001 3M02 606')
+            ->setCode(Bic::INVALID_IBAN_COUNTRY_CODE_ERROR)
+            ->assertRaised();
+    }
+
+    public function testValidComparisonToValue()
+    {
+        $constraint = new Bic(array('iban' => 'FR14 2004 1010 0505 0001 3M02 606'));
+        $constraint->ibanMessage = 'Constraint Message';
+
+        $this->validator->validate('SOGEFRPP', $constraint);
+
+        $this->assertNoViolation();
+    }
+
+    public function testInvalidComparisonToValue()
+    {
+        $constraint = new Bic(array('iban' => 'FR14 2004 1010 0505 0001 3M02 606'));
+        $constraint->ibanMessage = 'Constraint Message';
+
+        $this->validator->validate('UNCRIT2B912', $constraint);
+
+        $this->buildViolation('Constraint Message')
+            ->setParameter('{{ value }}', '"UNCRIT2B912"')
+            ->setParameter('{{ iban }}', 'FR14 2004 1010 0505 0001 3M02 606')
+            ->setCode(Bic::INVALID_IBAN_COUNTRY_CODE_ERROR)
+            ->assertRaised();
+    }
+
+    public function testNoViolationOnNullObjectWithPropertyPath()
+    {
+        $constraint = new Bic(array('ibanPropertyPath' => 'propertyPath'));
+
+        $this->setObject(null);
+
+        $this->validator->validate('UNCRIT2B912', $constraint);
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Validator\Exception\ConstraintDefinitionException
+     * @expectedExceptionMessage requires only one of the "iban" or "ibanPropertyPath" options to be set, not both.
+     */
+    public function testThrowsConstraintExceptionIfBothValueAndPropertyPath()
+    {
+        new Bic(array(
+            'iban' => 'value',
+            'ibanPropertyPath' => 'propertyPath',
+        ));
+    }
+
+    public function testInvalidValuePath()
+    {
+        $constraint = new Bic(array('ibanPropertyPath' => 'foo'));
+
+        if (method_exists($this, 'expectException')) {
+            $this->expectException(ConstraintDefinitionException::class);
+            $this->expectExceptionMessage(sprintf('Invalid property path "foo" provided to "%s" constraint', \get_class($constraint)));
```

Comment:
I'm wondering: what's the full message like? How much repetition does it contain, if any?

Reply:
@nicolas-grekas I've copied-pasted the code from https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Tests/Constraints/AbstractComparisonValidatorTestCase.php#L176

Reply:
I'd still be interested in the answer :)

Reply:
@nicolas-grekas Invalid property path "foo" provided to "\Symfony\Component\Validator\Constraints\Bic" constraint

Reply:
Sorry it looks like I wasn't explicit enough: on the line above that throws the exception, the message ends with [...] constaint`: %s`, where %s is replaced by some `$e->getMessage()`. My question is about the value of this %s message: what does it look like in this test case for example?

Reply:
@nicolas-grekas Neither the property "foo" nor one of the methods "getFoo()", "foo()", "isFoo()", "hasFoo()", "__get()" exist and have public access in class "Symfony\Component\Validator\Tests\Constraints\BicComparisonTest_Class".

Reply:
Neither the property "foo" nor one of the methods "getFoo()", "foo()", "isFoo()", "hasFoo()", "__get()" exist and have public access in class "Symfony\Component\Validator\Tests\Constraints\BicComparisonTest_Class"